db/fixtures/ae_datastore/ManageIQ/System/CommonMethods/QuotaMethods.class/__methods__/validate_quota.rb

```diff
@@ -0,0 +1,154 @@
+
+def request_info
+  @service = ($evm.root['vmdb_object_type'] == 'service_template_provision_task') ? true : false
+  @miq_request = $evm.root['miq_request']
+  $evm.log(:info, "Request: #{@miq_request.description} id: #{@miq_request.id} ")
+end
+
+def source_info
+  @source = $evm.root['quota_source']
+end
+
+def quota_check(item, used, requested, quota_max, quota_warn)
+  $evm.log(:info, "Item: #{item} Used: (#{used}) Requested: (#{requested}) Max: (#{quota_max}) Warn: (#{quota_warn})")
+  return unless quota_max + quota_warn > 0
+  if quota_exceeded?(item, used, requested, quota_max)
+    quota_exceeded(item, reason(item, used, requested, quota_max), false)
+  elsif quota_exceeded?(item, used, requested, quota_warn)
+    quota_exceeded(item, reason(item, used, requested, quota_warn), true)
+  end
+end
+
+def quota_exceeded?(item, used, requested, quota)
+  return false if quota.zero?
+  if used + requested > quota
+    $evm.log(:info, "#{item} Quota exceeded: Used(#{used}) + Requested(#{requested}) > Quota(#{quota})")
+    return true
+  end
+  false
+end
+
+def quota_exceeded(item, reason, warn)
+  warn ? quota_warn_exceeded(item, reason) : quota_limit_exceeded(item, reason)
+  true
+end
+
+def reason_key(item, warn)
+  "#{item}_quota_#{warn}exceeded".to_sym
+end
+
+def quota_limit_exceeded(item, reason)
+  key = reason_key(item, nil)
+  $evm.log(:info, "Quota maximum allowed exceeded for key: #{key} reason: #{reason}")
+  @max_exceeded[key] = reason
+end
+
+def quota_warn_exceeded(item, reason)
+  key = reason_key(item, "warn_")
+  $evm.log(:info, "Quota Warning limit exceeded for key: #{key} reason: #{reason}")
+  @warn_exceeded[key] = reason
+end
+
+def reason(item, used, requested, limits)
+  "#{item} - Used: #{used} plus requested: #{requested} exceeds quota: #{limits}"
+end
+
+def check_quotas
+  %w(storage vms cpu memory).each do |i|
+    key = i.to_sym
+    quota_check(i, @used[key].to_i, @requested[key].to_i, @max_limit[key].to_i, @warn_limit[key].to_i)
+  end
+end
+
+def check_quota_results
```

Comment:
@tinaafitz 
if there are 2 parameters (e.g. cpu and memory) that exceed the limits will we get the message for both them in one attempt or would it do 1 at a time. It would be preferred if we told them all the exceeded limits in a single attempt so that they can change all once.